src/core/ddsc/src/dds_topic.c

```diff
@@ -78,24 +78,21 @@ static dds_return_t dds_topic_status_validate (uint32_t mask)
 #ifdef DDS_HAS_TOPIC_DISCOVERY
 static struct ktopic_type_guid * topic_guid_map_refc_impl (const struct dds_ktopic * ktp, const struct ddsi_sertype *sertype, bool unref)
 {
-  type_identifier_t *tid = ddsi_typeid_from_sertype (sertype);
-  if (ddsi_typeid_none (tid))
-  {
-    // tid may be a null pointer but it may also be all-zero
-    ddsrt_free (tid);
+  ddsi_typeid_t *tid = ddsi_sertype_typeid (sertype, DDSI_TYPEID_KIND_COMPLETE);
+  if (ddsi_typeid_is_none (tid))
```

Comment:
`dds_create_topic_impl` now checks that, in case there is type information in the sertype, it has both a minimal and complete type id set: https://github.com/dennis-adlink/cyclonedds/blob/95f76dfc5be1a6c8c8f1963da9f64f9be94a60a9/src/core/ddsc/src/dds_topic.c#L437